lib/rucio/transfertool/fts3.py

```diff
@@ -531,7 +533,7 @@ def get_monitor_msg_fields(self, session, logger=logging.log):
     def _transfer_link(self):
         return '%s/fts3/ftsmon/#/job/%s' % (self._fts_address.replace('8446', '8449'), self._transfer_id)
 
-    def _find_attribute_updates(self, request, new_state, reason, overwrite_corrupted_files):
+    def _find_attribute_updates(self, request: dict, new_state: RequestState, reason: str, overwrite_corrupted_files: Optional[bool]):
```

Comment:
I would change `overwrite_corrupted_files: Optional[bool]` to `overwrite_corrupted_files: bool = False`

return type: `Optional[dict[str, Any]])`